src/test/java/org/aksw/simba/squirrel/iterators/SqlBasedIteratorTest.java

```diff
@@ -0,0 +1,83 @@
+package org.aksw.simba.squirrel.iterators;
+
+    import java.io.File;
+    import java.io.IOException;
+    import java.sql.Connection;
+    import java.sql.DriverManager;
+    import java.sql.PreparedStatement;
+    import java.sql.SQLException;
+    import java.util.regex.Pattern;
+
+    import org.aksw.simba.squirrel.utils.TempFileHelper;
+    import org.junit.AfterClass;
+    import org.junit.Assert;
+    import org.junit.BeforeClass;
+    import org.junit.Test;
+    import org.slf4j.Logger;
+    import org.slf4j.LoggerFactory;
+
+public class SqlBasedIteratorTest {
+    private static final Logger LOGGER = LoggerFactory.getLogger(SqlBasedIteratorTest.class);
+    private static Connection dbConnection;
+    private static final String CREATE_TABLE_QUERY = "CREATE TABLE ? (uri VARCHAR(255), serial INT, data BLOB, PRIMARY KEY(uri,serial));";
+
+    @BeforeClass
+    public static void createDb() throws SQLException, IOException {
+        File dbdir = TempFileHelper.getTempDir("dbTest", "");
+        System.out.println("using " + dbdir);
+        dbConnection = DriverManager.getConnection("jdbc:hsqldb:" + dbdir.getAbsolutePath() + File.separator + "test",
+            "SA", "");
+    }
+
+    @AfterClass
+    public static void closeDb() throws SQLException {
+        dbConnection.close();
+    }
+
+
+
+    @Test
+    public void test() throws SQLException {
+        PreparedStatement stmt;
+        String uri = "http://example.org/example";
+        String tableName = generateTableName(uri);
+        dbConnection.createStatement().executeUpdate(CREATE_TABLE_QUERY.replaceAll("\\?", tableName));
+        dbConnection.commit();
+        try {
+            stmt = dbConnection.prepareStatement("select * from "+ tableName +" OFFSET ? FETCH NEXT ? ROWS ONLY");
+            stmt.setInt(1,0);//1 specifies the first parameter in the query
+            stmt.setInt(2,3);
+            SqlBasedIterator obj = new SqlBasedIterator(stmt);
+            boolean has_Next =  obj.hasNext();
+            Assert.assertFalse(has_Next);
+        } catch (SQLException e) {
+            LOGGER.error("Exception while selecting the values from table");
+        }
+
```

Comment:
This test is ok but you are not testing the main functionality of the SQL iterator. Your test only makes sure that the iterator is able to handle an empty table.

You simply can create your own table and you don't have to reuse the complicated way to create a table name. The only important point is that the 3rd column of your result is a `BLOB` (because of https://github.com/varunmaitreya/Squirrel/blob/feature_Integration_test_case/src/main/java/org/aksw/simba/squirrel/iterators/SqlBasedIterator.java#L49).
Your test should generate a table (similar to what you are doing now) and insert data (At least 2 "pages" for the iterator, i.e., 10 items). After that, it should make sure that the iterator retrieves all the data that your select query should have returned.

Reply:
@MichaelRoeder Have created another test case to validate the retrieval from SqlBasedIterator especially the 3rd column(byte[]). Let me know your comments please.